Errors in codebase CI runs using MyPy 1.7.0

Recent `codebase` CI runs with `mypy == 1.7.0` are failing as follows:
```
Run mypy --version && mypy
mypy 1.7.0 (compiled: yes)
src/bokeh/colors/color.py:294: error: Unused "type: ignore" comment 
[unused-ignore]
                r, g, b = value         # type:ignore # https://github.com.../
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
src/bokeh/colors/color.py:297: error: Unused "type: ignore" comment 
[unused-ignore]
                r, g, b, a = value      # type:ignore # https://github.com.../
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
src/bokeh/embed/standalone.py:137: error: Unused "type: ignore" comment, use
narrower [overload-overlap] instead of [misc] code  [unused-ignore]
    def components(models: Model, wrap_script: bool = ..., # type: ignore[...
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
src/bokeh/embed/standalone.py:144: error: Unused "type: ignore" comment, use
narrower [overload-overlap] instead of [misc] code  [unused-ignore]
    def components(models: Sequence[Model], wrap_script: bool = ..., # typ...
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
src/bokeh/embed/standalone.py:151: error: Unused "type: ignore" comment, use
narrower [overload-overlap] instead of [misc] code  [unused-ignore]
    def components(models: dict[str, Model], wrap_script: bool = ..., # ty...
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
src/bokeh/server/tornado.py:432: error: Unused "type: ignore" comment 
[unused-ignore]
                    r[2]["bokeh_websocket_path"] = websocket_path # type: ...
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
Found 6 errors in 3 files (checked 742 source files)
Error: Process completed with exit code 1.
```
There are no errors reported using `mypy < 1.7`; I am using `mypy == 1.6.1` locally.

--- a/conda/environment-test-3.10.yml
+++ b/conda/environment-test-3.10.yml
@@ -64,7 +64,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=0.990
+    - mypy >=1.7
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
@@ -74,7 +74,7 @@ dependencies:
     - sphinx-favicon
     - sphinxext-opengraph
     # tests
-    - pandas-stubs ==1.5.2.221213
+    - pandas-stubs >=2.1
     - ruff ==0.0.285
     - types-boto
     - types-colorama

--- a/conda/environment-test-3.11.yml
+++ b/conda/environment-test-3.11.yml
@@ -64,7 +64,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=0.990
+    - mypy >=1.7
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
@@ -74,7 +74,7 @@ dependencies:
     - sphinx-favicon
     - sphinxext-opengraph
     # tests
-    - pandas-stubs ==1.5.2.221213
+    - pandas-stubs >=2.1
     - ruff ==0.0.285
     - types-boto
     - types-colorama

--- a/conda/environment-test-3.12.yml
+++ b/conda/environment-test-3.12.yml
@@ -64,7 +64,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=0.990
+    - mypy >=1.7
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
@@ -74,7 +74,7 @@ dependencies:
     - sphinx-favicon
     - sphinxext-opengraph
     # tests
-    #- pandas-stubs ==1.5.2.221213 # Not for python 3.12
+    - pandas-stubs >=2.1
     - ruff ==0.0.285
     - types-boto
     - types-colorama

--- a/conda/environment-test-3.9.yml
+++ b/conda/environment-test-3.9.yml
@@ -64,7 +64,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=0.990
+    - mypy >=1.7
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
@@ -74,7 +74,7 @@ dependencies:
     - sphinx-favicon
     - sphinxext-opengraph
     # tests
-    - pandas-stubs ==1.5.2.221213
+    - pandas-stubs >=2.1
     - ruff ==0.0.285
     - types-boto
     - types-colorama

--- a/conda/environment-test-minimal-deps.yml
+++ b/conda/environment-test-minimal-deps.yml
@@ -53,7 +53,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=0.990
+    - mypy >=1.7
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0

--- a/src/bokeh/colors/color.py
+++ b/src/bokeh/colors/color.py
@@ -291,10 +291,10 @@ class RGB(Color):
     def from_tuple(cls, value: RGBTuple) -> RGB:
         ''' Initialize ``RGB`` instance from a 3- or 4-tuple. '''
         if len(value) == 3:
-            r, g, b = value         # type:ignore # https://github.com/python/mypy/issues/1178
+            r, g, b = value
             return RGB(r, g, b)
         else:
-            r, g, b, a = value      # type:ignore # https://github.com/python/mypy/issues/1178
+            r, g, b, a = value
             return RGB(r, g, b, a)
 
     @classmethod

--- a/src/bokeh/embed/standalone.py
+++ b/src/bokeh/embed/standalone.py
@@ -134,21 +134,21 @@ def autoload_static(model: Model | Document, resources: Resources, script_path: 
     return js, tag
 
 @overload
-def components(models: Model, wrap_script: bool = ..., # type: ignore[misc] # XXX: mypy bug
+def components(models: Model, wrap_script: bool = ..., # type: ignore[overload-overlap] # XXX: mypy bug
     wrap_plot_info: Literal[True] = ..., theme: ThemeLike = ...) -> tuple[str, str]: ...
 @overload
 def components(models: Model, wrap_script: bool = ..., wrap_plot_info: Literal[False] = ...,
     theme: ThemeLike = ...) -> tuple[str, RenderRoot]: ...
 
 @overload
-def components(models: Sequence[Model], wrap_script: bool = ..., # type: ignore[misc] # XXX: mypy bug
+def components(models: Sequence[Model], wrap_script: bool = ..., # type: ignore[overload-overlap] # XXX: mypy bug
     wrap_plot_info: Literal[True] = ..., theme: ThemeLike = ...) -> tuple[str, Sequence[str]]: ...
 @overload
 def components(models: Sequence[Model], wrap_script: bool = ..., wrap_plot_info: Literal[False] = ...,
     theme: ThemeLike = ...) -> tuple[str, Sequence[RenderRoot]]: ...
 
 @overload
-def components(models: dict[str, Model], wrap_script: bool = ..., # type: ignore[misc] # XXX: mypy bug
+def components(models: dict[str, Model], wrap_script: bool = ..., # type: ignore[overload-overlap] # XXX: mypy bug
     wrap_plot_info: Literal[True] = ..., theme: ThemeLike = ...) -> tuple[str, dict[str, str]]: ...
 @overload
 def components(models: dict[str, Model], wrap_script: bool = ..., wrap_plot_info: Literal[False] = ...,

--- a/src/bokeh/server/tornado.py
+++ b/src/bokeh/server/tornado.py
@@ -429,7 +429,7 @@ class BokehTornado(TornadoApplication):
                 raise RuntimeError("Couldn't find websocket path")
             for r in app_patterns:
                 assert len(r) == 3 # TODO: handle two valued case as well
-                r[2]["bokeh_websocket_path"] = websocket_path # type: ignore[misc] # TODO: possible bug in mypy; pyright supports the assertion
+                r[2]["bokeh_websocket_path"] = websocket_path
 
             all_patterns.extend(app_patterns)
 

--- a/src/bokeh/util/serialization.py
+++ b/src/bokeh/util/serialization.py
@@ -359,7 +359,7 @@ def transform_array(array: npt.NDArray[Any]) -> npt.NDArray[Any]:
 
     return array
 
-def transform_series(series: pd.Series[Any] | pd.Index | pd.api.extensions.ExtensionArray) -> npt.NDArray[Any]:
+def transform_series(series: pd.Series[Any] | pd.Index[Any] | pd.api.extensions.ExtensionArray) -> npt.NDArray[Any]:
     ''' Transforms a Pandas series into serialized form
 
     Args:
@@ -374,7 +374,7 @@ def transform_series(series: pd.Series[Any] | pd.Index | pd.api.extensions.Exten
     # not checking for pd here, this function should only be called if it
     # is already known that series is a Pandas Series type
     if isinstance(series, pd.PeriodIndex):
-        vals = series.to_timestamp().values  # type: ignore
+        vals = series.to_timestamp().values
     else:
         vals = series.to_numpy()
     return vals

